hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/ReplicateContainerCommandHandler.java

```diff
@@ -66,12 +66,15 @@ public void handle(SCMCommand command, OzoneContainer container,
     final List<DatanodeDetails> sourceDatanodes =
         replicateCommand.getSourceDatanodes();
     final long containerID = replicateCommand.getContainerID();
+    final long deadline = replicateCommand.getDeadline();
 
     Preconditions.checkArgument(sourceDatanodes.size() > 0,
         "Replication command is received for container %s "
             + "without source datanodes.", containerID);
 
-    supervisor.addTask(new ReplicationTask(containerID, sourceDatanodes));
+    ReplicationTask task = new ReplicationTask(containerID, sourceDatanodes);
+    task.setDeadline(deadline);
+    supervisor.addTask(task);
```

Comment:
For HDDS-7620 I will likely need to add `term` to `ReplicationTask` to be able to check it before the task is executed.  I was wondering if we should pass the entire command to `ReplicationTask` instead of duplicating more and more fields.

Reply:
Yea let me look at this. There is a similar pattern in the ECReconstruction command, where we basically duplicate the entire command into another object which really only has getters the same as the command, so we might be able to make that go away too.